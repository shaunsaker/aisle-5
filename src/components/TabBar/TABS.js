const TABS = [
  {
    name: 'Home',
    iconName: 'shopping-cart',
    sceneKey: '_home',
  },
  {
    name: 'Predictions',
    iconName: 'assessment',
    sceneKey: '_predictions',
  },
  {
    name: 'Lists',
    iconName: 'assignment',
    sceneKey: '_lists',
  },
];

export default TABS;
