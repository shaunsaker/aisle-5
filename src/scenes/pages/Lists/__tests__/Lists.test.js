import React from 'react';
import renderer from 'react-test-renderer';

import { Lists } from '..';
import USER_LISTS from '../../../../mockData/userLists';
import USER_ITEMS from '../../../../mockData/userItems';
import TABS from '../tabs';

jest.mock('../../../../components/NetworkActivityIndicator', () => 'NetworkActivityIndicator');
jest.mock('../../../../components/TabBar', () => 'TabBar');

describe('Lists', () => {
  const spies = [];
  const dispatch = jest.fn();
  const selectedTab = TABS[0];

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Lists userLists={USER_LISTS} userItems={USER_ITEMS} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle onTabPress', () => {
      spies[0] = jest.spyOn(Lists.prototype, 'setActiveTab');
      const component = renderer.create(<Lists userLists={USER_LISTS} userItems={USER_ITEMS} />);
      const instance = component.getInstance();

      instance.onTabPress(selectedTab);

      expect(spies[0]).toHaveBeenCalledWith(selectedTab);
    });

    it('should handle setActiveTab', () => {
      const component = renderer.create(<Lists userLists={USER_LISTS} userItems={USER_ITEMS} />);
      const instance = component.getInstance();

      instance.setActiveTab(selectedTab);

      expect(instance.state.activeTab).toEqual(selectedTab);
    });
  });

  describe('actions', () => {
    it('should call onTabPress on tab button press', () => {
      spies[0] = jest.spyOn(Lists.prototype, 'onTabPress');
      const component = renderer.create(<Lists userLists={USER_LISTS} userItems={USER_ITEMS} />);
      const { root } = component;
      const targetComponent = root.findByProps({
        testID: `lists.headerTabs.button.${selectedTab.text}`,
      });

      targetComponent.props.handlePress();

      expect(spies[0]).toHaveBeenCalledWith(selectedTab);
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
