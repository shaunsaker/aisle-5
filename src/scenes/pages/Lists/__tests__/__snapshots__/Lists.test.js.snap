// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#EDE8CA",
        "flex": 1,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#08293A",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#08293A",
          "borderColor": null,
          "borderWidth": 0,
          "elevation": 12,
          "height": 60,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingHorizontal": 15,
          "paddingTop": 10,
          "shadowColor": "#212121",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 4,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#212121",
            "fontFamily": "ArchivoBlack-Regular",
            "fontSize": 24,
          },
          Object {
            "color": "#EDE8CA",
          },
        ]
      }
    >
      Lists
    </Text>
    <View
      style={
        Object {
          "bottom": 0,
          "justifyContent": "center",
          "position": "absolute",
          "right": 15,
          "top": 0,
        }
      }
    >
      <NetworkActivityIndicator />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#FFFFFF",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 51,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="lists.headerTabs.button.WEEKLY"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#212121",
                  "fontFamily": "Muli-Bold",
                  "fontSize": 15,
                  "textDecorationLine": "underline",
                },
                Object {
                  "color": "#D17623",
                },
              ]
            }
          >
            WEEKLY
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="lists.headerTabs.button.MONTHLY"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#212121",
                  "fontFamily": "Muli-Bold",
                  "fontSize": 15,
                  "textDecorationLine": "underline",
                },
                false,
              ]
            }
          >
            MONTHLY
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="lists.headerTabs.button.YEARLY"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#212121",
                  "fontFamily": "Muli-Bold",
                  "fontSize": 15,
                  "textDecorationLine": "underline",
                },
                false,
              ]
            }
          >
            YEARLY
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      bounces={false}
      contentContainerStyle={
        Object {
          "paddingBottom": 70,
        }
      }
      data={
        Array [
          Object {
            "id": "1",
            "name": "Beer",
            "quantity": 36288,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={Object {}}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          style={null}
        >
          <View
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingHorizontal": 15,
                  },
                  Object {
                    "height": 51,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#212121",
                    "fontFamily": "Muli-Bold",
                    "fontSize": 17,
                  }
                }
              >
                36288
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#212121",
                    "fontFamily": "Muli-SemiBold",
                    "fontSize": 17,
                  }
                }
              >
                 x Beer
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <TabBar />
</View>
`;
